File: app/src/main/java/de/sysitprep/app/data/model/Lernfeld.kt
package de.sysitprep.app.data.model

/**
 * Represents a Lernfeld (learning field) as defined by the IHK curriculum.
 */
data class Lernfeld(
    val number: Int,
    val title: String,
    val description: String,
    val fachrichtungen: List<Fachrichtung>,
    val stunden: Int
)

object LernfelderData {

    val allLernfelder: List<Lernfeld> = listOf(
        // ── Gemeinsame Lernfelder (LF 1–7) ──────────────────────────────────────
        Lernfeld(1, "Das Unternehmen und die eigene Rolle im Betrieb beschreiben",
            "Unternehmensorganisation, Berufsausbildung, Rechte und Pflichten, Betriebsklima",
            listOf(Fachrichtung.GEMEINSAM), 40),
        Lernfeld(2, "Arbeitsplätze nach Kundenwunsch ausstatten",
            "Hardware, Betriebssysteme, Softwarelizenzen, IT-Systeme auswählen und installieren",
            listOf(Fachrichtung.GEMEINSAM), 80),
        Lernfeld(3, "Clients in Netzwerke einbinden",
            "OSI-Modell, TCP/IP, IP-Adressierung, DHCP, DNS, WLAN, grundlegende Netzwerkkonfiguration",
            listOf(Fachrichtung.GEMEINSAM), 80),
        Lernfeld(4, "Schutzbedarfsanalyse im eigenen Arbeitsbereich durchführen",
            "IT-Sicherheit, Datenschutz (DSGVO), Bedrohungsanalyse, Schutzmaßnahmen",
            listOf(Fachrichtung.GEMEINSAM), 80),
        Lernfeld(5, "Software zur Verwaltung von Daten anpassen",
            "Datenbanken, SQL, Datenmodellierung, ER-Diagramme, Normalisierung",
            listOf(Fachrichtung.GEMEINSAM), 80),
        Lernfeld(6, "Serviceanfragen bearbeiten",
            "ITIL, Ticketsystem, First-Level-Support, SLA, Incident-Management",
            listOf(Fachrichtung.GEMEINSAM), 40),
        Lernfeld(7, "Cyber-physische Systeme ergänzen",
            "IoT, Sensoren, Aktoren, Mikrocontroller, Schnittstellen, Protokolle",
            listOf(Fachrichtung.GEMEINSAM), 80),

        // ── Systemintegration – spezifische Lernfelder (LF 8–12 SI) ─────────────
        Lernfeld(8, "Netzwerke und Dienste bereitstellen",
            "Serverinfrastruktur, Active Directory, DNS, DHCP-Server, Routing, Firewalls",
            listOf(Fachrichtung.SI), 80),
        Lernfeld(9, "Netzwerke und Dienste absichern",
            "VPN, Firewall-Konzepte, Verschlüsselung, PKI, Zertifikate, Penetrationstest",
            listOf(Fachrichtung.SI), 80),
        Lernfeld(10, "Clients administrieren und absichern",
            "Gruppenrichtlinien, MDM, Virenschutz, Backup-Strategien, Hardening",
            listOf(Fachrichtung.SI), 80),
        Lernfeld(11, "Komplexe IT-Systeme einrichten",
            "Virtualisierung, Cloud-Dienste, Hochverfügbarkeit, Storage, SAN/NAS",
            listOf(Fachrichtung.SI), 80),
        Lernfeld(12, "IT-Systeme in Betrieb nehmen",
            "Projektplanung, Migration, Rollout, Dokumentation, ITIL-Betrieb",
            listOf(Fachrichtung.SI), 40),

        // ── Anwendungsentwicklung – spezifische Lernfelder (LF 8–12 AE) ─────────
        Lernfeld(8, "Daten systemübergreifend bereitstellen",
            "REST-APIs, JSON/XML, Web Services, Schnittstellendesign",
            listOf(Fachrichtung.AE), 80),
        Lernfeld(9, "Informationssysteme und Netzwerke entwerfen",
            "Netzwerkarchitektur, Datenbankserver, Middleware, Microservices",
            listOf(Fachrichtung.AE), 80),
        Lernfeld(10, "Benutzerschnittstellen gestalten und entwickeln",
            "UI/UX-Design, Web-Frontend, Native Apps, Barrierefreiheit, Responsive Design",
            listOf(Fachrichtung.AE), 80),
        Lernfeld(11, "Funktionalität in Anwendungen realisieren",
            "OOP-Konzepte, Entwurfsmuster, Unit-Tests, Versionsverwaltung (Git)",
            listOf(Fachrichtung.AE), 80),
        Lernfeld(12, "Softwareprojekte durchführen",
            "Agiles Projektmanagement, Scrum, Kanban, Qualitätssicherung, DevOps",
            listOf(Fachrichtung.AE), 40)
    )

    fun getLernfelderForFachrichtung(fachrichtung: Fachrichtung): List<Lernfeld> {
        return allLernfelder.filter {
            it.fachrichtungen.contains(fachrichtung) ||
            it.fachrichtungen.contains(Fachrichtung.GEMEINSAM)
        }
    }
}

File: ExamForge-iOS/ExamForge/Models/Lernfeld.swift
import Foundation

// MARK: - Lernfeld

struct Lernfeld: Identifiable {
    let number: Int
    let title: String
    let description: String
    let fachrichtungen: [Fachrichtung]
    let stunden: Int

    var id: String { "\(number)-\(fachrichtungen.map(\.rawValue).joined())" }
}

// MARK: - Static Data

enum LernfelderData {
    static let allLernfelder: [Lernfeld] = [
        Lernfeld(number: 1,
                 title: "Das Unternehmen und die eigene Rolle im Betrieb beschreiben",
                 description: "Unternehmensorganisation, Berufsausbildung, Rechte und Pflichten, Betriebsklima",
                 fachrichtungen: [.GEMEINSAM], stunden: 40),
        Lernfeld(number: 2,
                 title: "Arbeitspl\u{00e4}tze nach Kundenwunsch ausstatten",
                 description: "Hardware, Betriebssysteme, Softwarelizenzen, IT-Systeme ausw\u{00e4}hlen und installieren",
                 fachrichtungen: [.GEMEINSAM], stunden: 80),
        Lernfeld(number: 3,
                 title: "Clients in Netzwerke einbinden",
                 description: "OSI-Modell, TCP/IP, IP-Adressierung, DHCP, DNS, WLAN, grundlegende Netzwerkkonfiguration",
                 fachrichtungen: [.GEMEINSAM], stunden: 80),
        Lernfeld(number: 4,
                 title: "Schutzbedarfsanalyse im eigenen Arbeitsbereich durchf\u{00fc}hren",
                 description: "IT-Sicherheit, Datenschutz (DSGVO), Bedrohungsanalyse, Schutzma\u{00df}nahmen",
                 fachrichtungen: [.GEMEINSAM], stunden: 80),
        Lernfeld(number: 5,
                 title: "Software zur Verwaltung von Daten anpassen",
                 description: "Datenbanken, SQL, Datenmodellierung, ER-Diagramme, Normalisierung",
                 fachrichtungen: [.GEMEINSAM], stunden: 80),
        Lernfeld(number: 6,
                 title: "Serviceanfragen bearbeiten",
                 description: "ITIL, Ticketsystem, First-Level-Support, SLA, Incident-Management",
                 fachrichtungen: [.GEMEINSAM], stunden: 40),
        Lernfeld(number: 7,
                 title: "Cyber-physische Systeme erg\u{00e4}nzen",
                 description: "IoT, Sensoren, Aktoren, Mikrocontroller, Schnittstellen, Protokolle",
                 fachrichtungen: [.GEMEINSAM], stunden: 80),
        // SI-specific
        Lernfeld(number: 8,
                 title: "Netzwerke und Dienste bereitstellen",
                 description: "Serverinfrastruktur, Active Directory, DNS, DHCP-Server, Routing, Firewalls",
                 fachrichtungen: [.SI], stunden: 80),
        Lernfeld(number: 9,
                 title: "Netzwerke und Dienste absichern",
                 description: "VPN, Firewall-Konzepte, Verschl\u{00fc}sselung, PKI, Zertifikate, Penetrationstest",
                 fachrichtungen: [.SI], stunden: 80),
        Lernfeld(number: 10,
                 title: "Clients administrieren und absichern",
                 description: "Gruppenrichtlinien, MDM, Virenschutz, Backup-Strategien, Hardening",
                 fachrichtungen: [.SI], stunden: 80),
        Lernfeld(number: 11,
                 title: "Komplexe IT-Systeme einrichten",
                 description: "Virtualisierung, Cloud-Dienste, Hochverf\u{00fc}gbarkeit, Storage, SAN/NAS",
                 fachrichtungen: [.SI], stunden: 80),
        Lernfeld(number: 12,
                 title: "IT-Systeme in Betrieb nehmen",
                 description: "Projektplanung, Migration, Rollout, Dokumentation, ITIL-Betrieb",
                 fachrichtungen: [.SI], stunden: 40),
        // AE-specific
        Lernfeld(number: 8,
                 title: "Daten system\u{00fc}bergreifend bereitstellen",
                 description: "REST-APIs, JSON/XML, Web Services, Schnittstellendesign",
                 fachrichtungen: [.AE], stunden: 80),
        Lernfeld(number: 9,
                 title: "Informationssysteme und Netzwerke entwerfen",
                 description: "Netzwerkarchitektur, Datenbankserver, Middleware, Microservices",
                 fachrichtungen: [.AE], stunden: 80),
        Lernfeld(number: 10,
                 title: "Benutzerschnittstellen gestalten und entwickeln",
                 description: "UI/UX-Design, Web-Frontend, Native Apps, Barrierefreiheit, Responsive Design",
                 fachrichtungen: [.AE], stunden: 80),
        Lernfeld(number: 11,
                 title: "Funktionalit\u{00e4}t in Anwendungen realisieren",
                 description: "OOP-Konzepte, Entwurfsmuster, Unit-Tests, Versionsverwaltung (Git)",
                 fachrichtungen: [.AE], stunden: 80),
        Lernfeld(number: 12,
                 title: "Softwareprojekte durchf\u{00fc}hren",
                 description: "Agiles Projektmanagement, Scrum, Kanban, Qualit\u{00e4}tssicherung, DevOps",
                 fachrichtungen: [.AE], stunden: 40)
    ]

    static func getLernfelderForFachrichtung(_ fachrichtung: Fachrichtung) -> [Lernfeld] {
        allLernfelder.filter { lf in
            lf.fachrichtungen.contains(.GEMEINSAM) || lf.fachrichtungen.contains(fachrichtung)
        }
    }
}
